applications/visualizer/frontend/src/components/ViewerSettings.tsx

```diff
@@ -52,9 +52,10 @@ const buttonStyle = {
   p: "0.25rem",
 };
 
-const ViewerSettings = ({ open, toggleDrawer }) => {
+const ViewerSettings = ({ open, toggleDrawer, workspaceId }) => {
   const dispatch = useDispatch();
-  const currentWorkspace = useGlobalContext().getCurrentWorkspace();
+  const { workspaces } = useGlobalContext();
+  const currentWorkspace = workspaces[workspaceId];
```

Comment:
Is it possible to get a new method or something special on the GlobalContext to access this workspace without having to get `workspaces`, it feels weird to have to get all the workspaces to only select one of them at the level of this component.